include/onnxruntime/core/providers/dml/dml_provider_factory.h

```diff
@@ -30,6 +30,23 @@
 extern "C" {
 #endif
 
+enum class OrtDmlPerformancePreference {
```

Comment:
>enum class [](http://example.com/codeflow?start=0&length=10)

`enum class` is not compatible with C, and this header file is C-includable (note lines 29-31). Unfortunately C enums leak their guts into the outer scope require prefixing.

```
enum class OrtDmlPerformancePreference {
  OrtDmlPerformancePreferenceDefault = 0,
  OrtDmlPerformancePreferenceHighPerformance = 1,
  OrtDmlPerformancePreferenceLowPower = 2
};
```

-------
Like in DirectML.h

```
enum DML_TENSOR_DATA_TYPE
{
    DML_TENSOR_DATA_TYPE_UNKNOWN,
    DML_TENSOR_DATA_TYPE_FLOAT32,
    DML_TENSOR_DATA_TYPE_FLOAT16,
...
```